### PR TITLE
Setup Leaflet and React Leaflet

### PR DESCRIPTION
**Pull request by MacAndersonUche**
- Install react-leaflet and leaflet in nextjs app.
- Creates a simple working example of a map 

References Issue: #8

### CODE REVIEW

**Review by ranegray — APPROVED** [review]

**Comment by toreylittlefield** [comment]
@MacAndersonUche can you reference the GH issue here?

**Comment by ranegray** [comment]
Working on #8. Added Map.tsx to pages folder as a component. I'm having trouble with Typescript errors because they aren't recognizing the react-leaflet component attributes. Tried installing @types/react-leaflet as per the docs but alas didn't fix the red squiggles.   The map will need to be centered still and markers will need to be ironed out. Let me know if there is anything glaringly breaking.

File: pages/index.tsx [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-import Image from 'next/image'
-import { Inter } from 'next/font/google'
+import dynamic from 'next/dynamic';
 import styles from '@/styles/Home.module.css'
 
-const inter = Inter({ subsets: ['latin'] })
+
+const MapWithNoSSR = dynamic(() => import('./Map'), {
+  ssr: false,
+});
```
Comment (toreylittlefield) [inline pages/index.tsx:8]: We can skip the dynamic import for now and keep it simple. Just import and render the component. Is this working for you? It wasn't for me.

**Comment by toreylittlefield** [comment]
For `Map.jsx` you can move it of the `pages` dir which is reserved in nextjs for `pages` (don't worry about it) and create a folder in the root of our project call `lib`. Inside lib create a `components` folder. Put `Map.jsx` inside components for now 👍

So importing `Map` should have the path `@/lib/components/Map`

Nextjs file structure is just a little different than React. Sorry I forgot to tell you.

File: pages/index.tsx (hunk 1 above)
Comment (MacAndersonUche) [reply pages/index.tsx:8]: I can confirm that it is not working for me

Comment (toreylittlefield) [reply pages/index.tsx:8]: Update: I read instructions on react-leaflet and nextjs ssr. Got it. This looks correct with the dynamic import like this for now. If we use SSR that'll be useful.

**Comment by toreylittlefield** [comment]
@ranegray I took a look at this and I apologize it is working for me. It's just I couldn't see the map. Can you adjust the style to make the map fit the entire viewport?

File: next.config.js [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+        patterns: [
+          {
+            from: 'node_modules/leaflet/dist/images',
+            to: path.resolve(__dirname, 'public', 'leaflet', 'images')
+          },
+        ],
+      }),
+    )
+    return config
+  }
```
Comment (toreylittlefield) [inline next.config.js:19]: @ranegray @MacAndersonUche why do we need this? Can we load the images from a CDN instead?

File: pages/Map.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from "react";
+import "leaflet/dist/leaflet.css";
+import { MapContainer, TileLayer, Marker, useMap } from "react-leaflet";
+
+export function ChangeView({ coords }) {
+  const map = useMap();
+  map.setView(coords, 12);
+  return null;
```
Comment (toreylittlefield) [inline pages/Map.jsx:8]: @ranegray are we using this? If not let's remove it for now.

File: pages/Map.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { useState, useEffect } from "react";
+import "leaflet/dist/leaflet.css";
+import { MapContainer, TileLayer, Marker, useMap } from "react-leaflet";
```
Comment (toreylittlefield) [inline pages/Map.jsx:3]: remove any unused imports

File: pages/Map.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+export function ChangeView({ coords }) {
+  const map = useMap();
+  map.setView(coords, 12);
+  return null;
+}
+
+export default function Map() {
```
Comment (toreylittlefield) [inline pages/Map.jsx:11]: let's use `const` instead of `function` just preference 👍

Comment (ranegray) [reply pages/Map.jsx]: changed to const in newest commit. export default at bottom of file.

File: next.config.js (hunk 2 above)
Comment (ranegray) [reply next.config.js:19]: Left this config in. Not sure if you wanted me to remove it.

**Comment by ranegray** [comment]
Newest commit. Added full viewport map, added lib/component file structure. Changed naming convention of function to arrow sytanx with default export at bottom of Maps.jsx. Removed any imports or functions that weren't being used. 

Clean slate ready for whatever.

File: next.config.js (hunk 2 above)
Comment (toreylittlefield) [reply next.config.js:19]: Yes looks like we need it for now 👍

File: lib/Components/Map.jsx [hunk 6]
```diff
@@ -17,7 +11,7 @@ export default function Map() {
     <MapContainer
       center={center}
       zoom={12}
-      style={{ height: "50%", width: "100%" }}
+      style={{ height: "100vh", width: "100vw" }}
```
Comment (toreylittlefield) [inline lib/Components/Map.jsx:14]: @ranegray for `width` you don't need to specify the viewport units as the cause problems👍 100% is good.

**Review by toreylittlefield — APPROVED** [review]
@ranegray well done 🚀

**Pull request merged**